packages/openactive-broker-microservice/src/util/persistent-store.js

```diff
@@ -0,0 +1,488 @@
+const { CriteriaOrientedOpportunityIdCache } = require('./criteria-oriented-opportunity-id-cache');
+const { mapToObjectSummary } = require('./map-to-object-summary');
+
+/**
+ * @typedef {Record<string, unknown>} OpportunityCacheItem
+ */
+
+/**
+ * Manages Broker Microservice data that scales with feed size, so contains
+ * caches of opportunity data.
+ *
+ * As feeds can be very large, caches which contain all feed data need to (at
+ * least optionally) be persisted to something other than memory.
+ *
+ * So this class is an abstraction over the persistence layer.
+ *
+ * Note that all get- functions return readonly data to ensure that client code
+ * does not attempt to mutate it, as this likely means that it is under the
+ * misinterpretation that this will update the data itself (as if it was
+ * in-memory).
+ */
+class PersistentStore {
+  constructor() {
+    /**
+     * A criteria-oriented cache for opportunity data. Used to get criteria-matching
+     * opportunities for tests.
+     */
+    this._criteriaOrientedOpportunityIdCache = CriteriaOrientedOpportunityIdCache.create();
```

Comment:
All of these values and their docs copied over from state.js (and with the `_` prefix added - making them super private 🕵️ )